### PR TITLE
[JNI] Adds HostColumnVector.EventHandler for spillability checks

### PR DESCRIPTION
**Pull request by abellina**
This change adds the exact same API for event handling that was added to the device-side `ColumnVector` here https://github.com/rapidsai/cudf/pull/13279. We are going to need this to make `HostColumnVector`, or a batch of them,  spillable in this spark-rapids feature: https://github.com/NVIDIA/spark-rapids/issues/8882.

### CODE REVIEW

**Comment by abellina** [comment]
Tests pass for me locally. I am seeing a CI failure related to a missing nvtx3 header, but I wasn't upmerged. Trying again:

```
     [exec] /opt/conda/envs/test/include/cudf/detail/nvtx/nvtx3.hpp:33:10: fatal error: nvtx3/nvToolsExt.h: No such file or directory
```

**Comment by abellina** [comment]
Failed again, cuDF jni isn't compiling in CI, but it runs fine locally :( It seems the `nvToolsExt` header is no longer available in the conda environment...?

File: java/src/main/java/ai/rapids/cudf/HostColumnVector.java
Comment (ttnghia) [inline java/src/main/java/ai/rapids/cudf/HostColumnVector.java:55]:
```suggestion
     * @param cv Reference to the HostColumnVector we are closing
     * @param refCount The updated ref count for this HostColumnVector at
     *        the time of invocation
```

File: java/src/main/java/ai/rapids/cudf/HostColumnVector.java
Comment (ttnghia) [inline java/src/main/java/ai/rapids/cudf/HostColumnVector.java:51]:
```suggestion
     * @note The callback is invoked with this `HostColumnVector`'s lock held.
```

**Comment by abellina** [comment]
I am trying a suggestion from @robertmaynard here https://github.com/rapidsai/cudf/pull/13898/commits/6fb3f1b8a702aa03bec65f7163592cb6651273cd to get the conda environment setup for the java test run with cuda 12, which is the one that is failing because the nvtx header we need is in a different conda package now.

File: dependencies.yaml
Comment (revans2) [inline dependencies.yaml:546]: where did this come from?

**Review by gerashegalov — APPROVED** [review]
LGTM

**Review by revans2 — COMMENTED** [review]
Looks good, just curious about the deps.

**Review by revans2 — APPROVED** [review]

File: java/src/main/java/ai/rapids/cudf/HostColumnVector.java
Comment (abellina) [reply java/src/main/java/ai/rapids/cudf/HostColumnVector.java:51]: We tend to favor starting with lowercase and we also favor indenting the description so that it reads as one block, but I do think the hyphen is discouraged. How about a compromise? (https://github.com/rapidsai/cudf/pull/13898/commits/a0944eebf501006f9f542e09a9dfc1934dbdf4b2)

Comment (ttnghia) [reply java/src/main/java/ai/rapids/cudf/HostColumnVector.java:51]: It's not very important since we are not using the generated docds. But if we are the users who read the generated docs, we will see they are ugly with bad style: Sentences start without first letter being Capped.

Comment (ttnghia) [reply java/src/main/java/ai/rapids/cudf/HostColumnVector.java:51]: and sentences starting without cap. letter is not good IMO.

Comment (abellina) [reply java/src/main/java/ai/rapids/cudf/HostColumnVector.java:51]: Since our code styles on this are all over the place, I am going to suggest filing a follow on if you feel strongly about it. We should then in that issue fix all the comments and make sure that the pre commit hooks exercise them, so we have something consistent and easy to test.

**Comment by abellina** [comment]
@robertmaynard that worked! thanks! Also I do need an approval from `ops-codeowners`

**Comment by abellina** [comment]
@raydouglass would you mind taking a look at this from the ops-codeowner perspective?  The change that prompted this review request was: https://github.com/rapidsai/cudf/pull/13898/commits/6fb3f1b8a702aa03bec65f7163592cb6651273cd

**Comment by abellina** [comment]
I'll wait for https://github.com/rapidsai/cudf/pull/13901/files#diff-5475a6d76de4c506ee92cf6f941bba30a6a07b5881cfea531d42a6ec035095a6 then re-test and rollback my change here likely.

**Comment by abellina** [comment]
Removed `ops-codeowners` because this patch no longer has diffs in dependencies.yaml, it's all java files.

**Pull request merged**
